toplevel/opttopmain.ml

```diff
@@ -204,11 +233,21 @@ module Options = Main_args.Make_opttop_options (struct
   let _open s = open_modules := s :: !open_modules
   let _plugin p = Compplugin.load p
 
+  let _args = wrap_expand Arg.read_arg
+  let _args0 = wrap_expand Arg.read_arg0
+
   let anonymous = file_argument
 end);;
 
 let main () =
   native_code := true;
-  Arg.parse Options.list file_argument usage;
+  let list = ref Options.list in
+  begin
+    try
+      Arg.parse_and_expand_argv_dynamic current argv list file_argument usage;
+    with
+    | Arg.Bad msg -> Format.fprintf Format.err_formatter "%s%!" msg; exit 2
+    | Arg.Help msg -> Format.fprintf Format.std_formatter "%s%!" msg; exit 0
```

Comment:
It is the same as in `Arg.parse` which was called before. For the other expand options it is possible to use `Arg.parse_expand` which does make the array and the current visible to the outside.